pkg/controller/perconaxtradbcluster/perconaxtradbcluster_controller.go

```diff
@@ -214,7 +214,13 @@ func (r *ReconcilePerconaXtraDBCluster) deploy(cr *api.PerconaXtraDBCluster) err
 	if cr.Spec.Platform != nil {
 		serverVersion.Platform = *cr.Spec.Platform
 	}
-
+	if len(cr.Spec.SSLSecretName) > 0 {
+		cr.Spec.PXC.SSLSecretName = cr.Spec.SSLSecretName
+		cr.Spec.ProxySQL.SSLSecretName = cr.Spec.SSLSecretName
+	} else {
+		cr.Spec.PXC.SSLSecretName = cr.Name + "-ssl"
+		cr.Spec.ProxySQL.SSLSecretName = cr.Name + "-ssl"
```

Comment:
`cr.Spec.ProxySQL` can be `nil` so in case `Spec.ProxySQL` missing in CR this code will result in "nil pointer dereference". The same above in line 219.